### PR TITLE
vendor colorama

### PR DESCRIPTION
**Pull request by amogkam**
Remove colorama dependency by vendoring it.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by richardliaw** [comment]
breaks all the tests :)

File: python/ray/__init__.py [hunk 1]
```diff
@@ -74,6 +74,12 @@ def _configure_system():
 __commit__ = "{{RAY_COMMIT_SHA}}"
 __version__ = "2.0.0.dev0"
 
+# Vendored packages.
+# We have to set this before we impor the raylet.
```
Comment (amogkam) [inline python/ray/__init__.py:78]:
```suggestion
# We have to set this before we import the raylet.
# raylet depends on colorama
```

**Comment by simon-mo** [comment]
@amogkam why do we vendor by copy all the code? Can we just install it to `ray.thirdparty`? Similar to `pstuila dn setproctitle`. https://github.com/ray-project/ray/pull/7031

**Comment by simon-mo** [comment]
Do we need to modify the colorama source code?

**Comment by amogkam** [comment]
Ah yes you're totally right @simon-mo. Let me do that in a separate PR

**Comment by richardliaw** [comment]
Hmm, wait quick question, we can just do the Simon suggested action in this PR right?

**Comment by amogkam** [comment]
Closing this in favor of https://github.com/ray-project/ray/pull/17183

**Pull request closed without merging**